docs/debugging-dotnet-lambda-functions.md

```diff
@@ -0,0 +1,149 @@
+    * Replace `5679` with the port that you specified in `launch.json`.
+
+    ```bash
+    sam build # TODO: First set the environment variable that enables building the Debug configuration.
+    sam local invoke HelloWorldFunction -d 5679 --debugger-path path/to/project/.vsdbg --no-event
+    ```
+
+3. When you see `Waiting for debugger to attach...`, go back to Visual Studio Code and press F5 to attach the debugger to the handler that you invoked in the previous step.
+
+## Optional: Automatically start debugging when ready
+
+With the above steps, you need to manually invoke SAM CLI from the command line, wait for it to be ready, then attach the debugger. We can automate the process of invoking SAM CLI and waiting for it to be ready by using a `preLaunchTask`.
+
+1. Open `<sam app root>/.vscode/tasks.json` (create a new file if it does not already exist).
+2. Add the following contents to `tasks.json`:
+
+    * Replace `HelloWorldFunction` with the function that you wish to debug.
+    * Replace `path/to/project` with the relative path to the directory containing the `.csproj` project that you wish to debug.
+
+    ```jsonc
+    {
+        "version": "2.0.0",
+        "tasks": [
+            {
+                "label": "Debug .NET Core Lambda Function",
+                "type": "shell",
+                "command": "sam",
+                "args": [
+                    "local",
+                    "invoke",
+                    "HelloWorldFunction",
+                    "-d",
+                    "5679",
+                    "--debugger-path",
+                    "path/to/project/.vsdbg",
+                    "--no-event"
+                ],
+                "isBackground": true,
+                "problemMatcher": {
+                    "pattern": [
+                        {
+                            // Use regex that never matches anything.
+                            "regexp": "^(x)(\\b)(x)$",
+                            "file": 1,
+                            "location": 2,
+                            "message": 3
+                        }
+                    ],
+                    "background": {
+                        // This is how the debugger knows when it can attach
+                        "activeOnStart": true,
+                        "beginsPattern": "^Fetching lambci\\.\\w+ Docker container image",
+                        "endsPattern": "^Waiting for the debugger to attach\\.\\.\\.$"
+                    }
+                }
+            }
+        ]
+    }
+    ```
+
+3. Open `<sam app root>/.vscode/launch.json`, and add the following property to the `.NET Core: Remote Attach` configuration that you created earlier, after `"request": "attach",`:
+
+    ```jsonc
+    "preLaunchTask": "Debug .NET Core Lambda Function",
+    ```
+
+Now you can just press `F5`, and Visual Studio Code will invoke SAM CLI and wait for the `Waiting for debugger to attach...` message before attaching the debugger.
```

Comment:
It might be worth mentioning that you will have to run sam build between iterations (and that the env var is required too)

Reply:
✔️